README.md

```diff
@@ -191,20 +200,72 @@ When working in the context of multiple JS global objects (e.g., with a same-ori
 
 It's possible to get around this by passing `Box.prototype.deref()` to that other global object. This method "grants the right" to dereference boxes created in the context of that global object. In this way, different global objects are a little bit isolated from each other, when it comes to `Box`.
 
-###### Membrane/Ocap implications
+###### Membrane/Ocap analysis
+
+Our hope was that, the fact that object operations are used to access and call `Box.prototype.deref`, and this can be membrane-wrapped, would imply that the Box system would work great for membrane-based security--just run each membrane-enclosed piece of code in a separate Realm, and you can safely share Boxes, granting or denying access based on the access to the `Box` constructor or `deref` method.
+
+However, in systems which want to use membrane-based isolation *within* the same Realm, the whole feature would have to be turned off. This is to prevent boxes from "piercing" through membranes: a box primitive could be passed from one side of the membrane to the other (the membrane has no chance to intervene since boxes are not objects), and then it can simply be dereferenced. This would constitute an unmediated cross-membrane communication channel, going against the goals of the membrane system.
```

Comment:
You can fix this by returning a new instance of `Box` when the original passes through a membrane, and you wrap the derefed value in the new `Box`.

Reply:
How would the membrane do this? cc @erights

Reply:
You have the membrane create a second `Box` instance, and return that. You don't need to worry about identity, because you can't directly compare a value on one side to a value on the other. So having a new identity on other side doesn't matter.

```js
const membrane = new Proxy({}, {
  get(target, prop, receiver) {
    const value = Reflect.get(target, prop, receiver);
    if (typeof value === 'box') {
      return new Box(wrapInProxyIfNeeded(value.deref()));
    }
    // …
  }
});
```

The only time you have to worry if there is some builtin box instance as part of the normal global object. In that case, identity could be observed (by comparing the value returned from the proxy to the global box). But that seems very unlikely to happen.

Reply:
I guess if a Box is directly returned, that's the easy path; what if it's hidden deep in a Record or Tuple?

Reply:
It would have to be deeply replaced with a second instance.

But perhaps we could allow Proxies to wrap Records? Records are "primitive", but they act just like objects. That would allow membranes to use the lazy approach they currently use for objects.

Reply:
Primitives have the axiom that `Object(primitive)` returns a boxed version. `Object()` of a Proxy instance should return something `===` the instance - so if you could have a proxy of any primitive, then `typeof` on the Proxy would have to match, but also `Object()` would have to, and also autoboxing behavior would have to dynamically look up the right prototype object in the current realm (the last one is probably the hardest to figure out).

Reply:
Proxy doesn't support structural === like Records and Tuples do.

This patch is going to land soon, and the comments here won't be very visible. We should move discussion to issues.